cmd/acr/purge.go

```diff
@@ -463,43 +468,41 @@ func getManifestsToDelete(ctx context.Context, acrClient api.AcrCLIClientInterfa
 		return nil, err
 	}
 	// This will act as a set if a key is present then it should not be deleted because it is referenced by a multiarch manifest
-	// that will not be deleted
-	doNotDelete := map[string]bool{}
-	candidatesToDelete := []acr.ManifestAttributesBase{}
-	// Iterate over all manifests to discover multiarchitecture manifests
+	// or the manifest has subjects attached
+	doNotDelete := set.New[string]()
+	var candidatesToDelete []acr.ManifestAttributesBase
 	for resultManifests != nil && resultManifests.ManifestsAttributes != nil {
 		manifests := *resultManifests.ManifestsAttributes
 		for _, manifest := range manifests {
-			if (*manifest.MediaType == manifestListContentType || *manifest.MediaType == manifestIndexContentType) && manifest.Tags != nil {
-				// If a manifest list is found and it has tags then all the dependent digests are
-				// marked to not be deleted.
-				var manifestListBytes []byte
-				manifestListBytes, err = acrClient.GetManifest(ctx, repoName, *manifest.Digest)
+			var cm customManifest
+			if _, ok := manifestMediaTypes[*manifest.MediaType]; ok {
```

Comment:
nit: And this can be 
```suggestion
			if manifestMediaTypes.Contains(*manifest.MediaType) {
```

Reply:
done